bulls_and_cows/lib/bulls_and_cows.ex

```diff
@@ -0,0 +1,158 @@
+defmodule BullsAndCows do
+  use Accessible
+
+  @moduledoc """
+  A game of bulls and cows, rules:
+  You need to guess 4 different numbers in definite order.
+  If number is correct but in wrong position, its a cow, if value and position is correct, its a bull.
+
+  Game should be played in iex. First you init game by new_game/0,
+  then you should pass state to make_turn/2 until you win
+
+  Game example:
```

Comment:
Examples are better to have as [doctests](https://hexdocs.pm/ex_unit/ExUnit.DocTest.html#doctest/2).

Here I see `number_encrypted` shown as `String.t` while in struct definition it’s a list. This inconsistency would not be possible when one uses real code in doctests.